qt/widgets/instrumentview/src/GLDisplay.cpp

```diff
@@ -74,11 +74,19 @@ void GLDisplay::initializeGL() {
   // Set the relevant OpenGL rendering options
   setRenderingOptions();
   glViewport(0, 0, width(), height());
+}
 
-  // Clear the memory buffers
+/**
+ * Resets the buffer to the current background color.
+ * The caller is responsible for ensuring the GL buffer is current
+ */
+void GLDisplay::resetBackgroundColor() {
   QColor bgColor = currentBackgroundColor();
   glClearColor(GLclampf(bgColor.red() / 255.0), GLclampf(bgColor.green() / 255.0), GLclampf(bgColor.blue() / 255.0),
                1.0);
+  // Since we're manually manipulating the current GL buffer (as we're double buffered)
+  // we need to ask Qt to redraw, else we get a corrupted display with an empty surface
+  updateGL();
```

Comment:
Nice. I think you've cleared up a long-standing bug with the display being empty before the view is manipulated in some cases.